### PR TITLE
add dark theme support

### PR DESCRIPTION
**Pull request by jlaehne**
Adds dark theme support based on the pillow theme: https://pillow.readthedocs.io/en/stable/

Logo background is set to black for both dark and light themes.

Some first additions to the documentation, incorporate existing pages in the main folder (changelog, contributing guide).

### CODE REVIEW

**Comment by jlaehne** [comment]
See https://lumispy--103.org.readthedocs.build/en/103/citing.html to check out the docs. In Firefox use `Ctrl-Shift-I` to activate the development tools. Under 'Inspector' you can switch between dark (moon icon) and light (sun icon) modes.

**Review by jordiferrero — APPROVED** [review]

**Comment by jordiferrero** [comment]
Thanks for this.
Do we need to include the changelog in the documentation so visible? I followed the hyperspy documentation and did not see it there. I would consider removing it.

**Comment by jlaehne** [comment]
> Thanks for this. Do we need to include the changelog in the documentation so visible? I followed the hyperspy documentation and did not see it there. I would consider removing it.

I would like to have it in the docs, but it does not need to be so prominent.

In HyperSpy, it is the second point in the User Guide after introduction: https://hyperspy.org/hyperspy-doc/current/index.html

I could also see it in the `API References` section instead of the user guide? What do you think?

> Otherwise, feel free to merge :-)
I'll wait for the decision on .rst vs .md - in case we opt for .md I would revert the files that I have converted back to .md to avoid a back and forth, but leave the rest of the conversion for a separate PR.

**Comment by ericpre** [comment]
As a library user, I find comprehensive change log very useful and important!

File: .github/CONTRIBUTING.md
Comment (ericpre) [inline .github/CONTRIBUTING.md:1]: This file is used by github - see https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors and https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/encouraging-helpful-contributions-to-your-project-with-labels.

From a quick at the github documentation, rst is not mentioned anywhere, do we know if github can parse the `rst` format? Maybe simply add a link to a markdown file located at the right place, as there are alternatives. Using the https://github.com/LumiSpy/.github would have the advantage of keeping this repository tidy.

**Review by ericpre — APPROVED** [review]

File: .github/CONTRIBUTING.md
Comment (jlaehne) [reply .github/CONTRIBUTING.md:1]: Should be doable with the Markdown extension for Sphinx: https://www.sphinx-doc.org/en/master/usage/markdown.html

Then the files in the main folder would remain .md and could still be integrated in the docs. I'll check that out later today.

**Comment by jlaehne** [comment]
> As a library user, I find comprehensive change log very useful and important!

Any thought on putting it in the `API references` section @ericpre?

File: .github/CONTRIBUTING.md
Comment (jlaehne) [reply .github/CONTRIBUTING.md:1]: But, `rst` would be supported by GitHub as well: https://docs.github.com/en/repositories/working-with-files/using-files/working-with-non-code-files#rendering-differences-in-prose-documents

**Comment by ericpre** [comment]
I wouldn't put it in the API reference. In my opinion, the best location will be at the same level than `User guide`, `API reference`, etc., as this is for done for [scipy](https://scipy.github.io/devdocs/index.html).

On a related topic, I really like the [pydata-sphinx-theme](https://pydata-sphinx-theme.readthedocs.io/en/latest/), because I find it very clear and functional.

File: .github/CONTRIBUTING.md
Comment (ericpre) [reply .github/CONTRIBUTING.md:1]: OK, that would be brilliant! Then, once this is merged, github should show us a notification when making a new PR, because it should detect that the contributing guide has changed.

**Comment by jlaehne** [comment]
Not so easy to integrate `.md` as I thought, so only the files not imported to the docs remain in markdown.

Already started out on getting docstrings corrected so that they are properly parsed in the api documentation, but there are still a few issues remaining (added it to the todo in #100).

I'll merge now so that we can continue in new PRs.

**Pull request merged**
